arangod/Cluster/ClusterComm.cpp

```diff
@@ -332,13 +326,29 @@ void ClusterComm::cleanup() {
 ////////////////////////////////////////////////////////////////////////////////
 
 void ClusterComm::startBackgroundThread() {
-  _backgroundThread = new ClusterCommThread();
+  _roundRobin = 0;
 
-  if (!_backgroundThread->start()) {
-    LOG_TOPIC(FATAL, Logger::CLUSTER)
-      << "ClusterComm background thread does not work";
-    FATAL_ERROR_EXIT();
-  }
+  for(unsigned loop=0; loop<(TRI_numberProcessors()/8+1); ++loop) {
+    ClusterCommThread * thread = new ClusterCommThread();
```

Comment:
The memory will leak in case the `push_back` in line 335 fails.
Doesn't really matter here, because we will abort the program anyway. But in general we should rather use smart pointers than raw `new`/`delete` operations.